stdlib/public/runtime/BuiltinProtocolConformances.cpp

```diff
@@ -0,0 +1,810 @@
+              (" TUPLE_COMPARABLE_CONF ")) - 28) + 1\n"
+  #elif defined(__MACH__)
+  "  .long " COMPARABLE_LT_METHOD_DESCRIPTOR "@GOTPCREL + 5\n"
+  #endif
+  // This is a direct relative reference to the less than witness defined below.
+  "  .long ((" TUPLE_COMPARABLE_LESSTHAN ") - (" TUPLE_COMPARABLE_CONF ")) - 32\n"
+  #if defined(__ELF__)
+  // This is an indirectable relative reference to the GOT equivalent for the
+  // Comparable.<= method descriptor, hence why we add 1 to indicate
+  // indirect.
+  "  .long ((got." COMPARBALE_LTE_METHOD_DESCRIPTOR " - \
+              (" TUPLE_COMPARABLE_CONF ")) - 36) + 1\n"
+  #elif defined(__MACH__)
+  "  .long " COMPARBALE_LTE_METHOD_DESCRIPTOR "@GOTPCREL + 5\n"
+  #endif
+  // This is a direct relative reference to the less than or equal witness
+  // defined below.
+  "  .long ((" TUPLE_COMPARABLE_LESSTHANOREQUAL ") - \
+              (" TUPLE_COMPARABLE_CONF ")) - 40\n"
+  #if defined(__ELF__)
+  // This is an indirectable relative reference to the GOT equivalent for the
+  // Comparable.>= method descriptor, hence why we add 1 to indicate
+  // indirect.
+  "  .long ((got." COMPARABLE_GTE_METHOD_DESCRIPTOR " - \
+              (" TUPLE_COMPARABLE_CONF ")) - 44) + 1\n"
+  #elif defined(__MACH__)
+  "  .long " COMPARABLE_GTE_METHOD_DESCRIPTOR "@GOTPCREL + 5\n"
+  #endif
+  // This is a direct relative reference to the greater than or equal witness
+  // defined below.
+  "  .long ((" TUPLE_COMPARABLE_GREATERTHANOREQUAL ") - \
+              (" TUPLE_COMPARABLE_CONF ")) - 48\n"
+  #if defined(__ELF__)
+  // This is an indirectable relative reference to the GOT equivalent for the
+  // Comparable.> method descriptor, hence why we add 1 to indicate
+  // indirect.
+  "  .long ((got." COMPARABLE_GT_METHOD_DESCRIPTOR " - \
+              (" TUPLE_COMPARABLE_CONF ")) - 52) + 1\n"
+  #elif defined(__MACH__)
+  "  .long " COMPARABLE_GT_METHOD_DESCRIPTOR "@GOTPCREL + 5\n"
+  #endif
+  // This is a direct relative reference to the greater than witness defined
+  // below.
+  "  .long ((" TUPLE_COMPARABLE_GREATERTHAN ") - \
+              (" TUPLE_COMPARABLE_CONF ")) - 56\n"
+  // The witness table size in words.
+  "  .short 0\n"
+  // The witness table private size in words & requires instantiation.
+  "  .short 1\n"
+  // The witness table instantiator function.
+  "  .long 0\n"
+  // This is a direct relative reference to the private data for the
+  // conformance.
+  "  .long (__swift_tupleComparable_private - (" TUPLE_COMPARABLE_CONF ")) - 68\n"
+  #if defined(__ELF__)
+  "  .size " TUPLE_COMPARABLE_CONF ", 72\n"
+  #endif
+);
+
+SWIFT_RUNTIME_EXPORT SWIFT_CC(swift)
+const WitnessTable *
+_swift_tupleComparable_baseAccessorEquatable(Metadata *assocType,
+                                             Metadata *conformingType,
+                                             void **witnessTable) {
+  auto tuple = cast<TupleTypeMetadata>(assocType);
+  std::vector<void *> instantiationArgs;
```

Comment:
We could `reserve` capacity for the instantiationArgs here, to reduce vector growth overhead.